src/main/scala/it/pagopa/interop/authorizationserver/api/impl/AuthApiServiceImpl.scala

```diff
@@ -71,10 +70,10 @@ final case class AuthApiServiceImpl(
       clientUUID             <- clientId.traverse(_.toUUID)
       clientAssertionRequest <- ValidClientAssertionRequest
         .from(clientAssertion, clientAssertionType, grantType, clientUUID)
-        .recoverWith { case err: InvalidAccessTokenRequest => Failure(InvalidAssertion(err.errors.mkString(","))) }
+        .adaptError { case err: InvalidAccessTokenRequest => InvalidAssertion(err.errors.mkString(",")) }
```

Comment:
It's better to do not use cats methods on Future, we can't safely say where this adaptError is going to be executed. To get this syntax a typeclass instance is instantiated using an implicit ec, and we can't safely say where it comes from. In general, using cats syntax on Future should be avoided.

Reply:
👌

Reply:
I approve this PR, but this comment is still open, isn't it?

Reply:
I removed all other occurrences, but this is used on `Try`. It should be safe, right?